MAVProxy/mavproxy.py

```diff
@@ -391,7 +391,11 @@ def unload_module(modname):
     for (m,pm) in mpstate.modules:
         if m.name == modname:
             if hasattr(m, 'unload'):
-                m.unload()
+                t = threading.Thread(target=lambda : m.unload(), name="unload %s" % modname)
+                t.start()
+                t.join(timeout=1)
+                if t.isAlive():
+                    print("unload on module %s did not complete" % m.name)
```

Comment:
Shouldn't we kill the thread and return False here?

Reply:
That's unclear to me.  I guess we can always add that later.

But I have increased the timeout to 5 seconds and returned false here after discussions with tridge.